libtsuba/src/RDGPartHeader.h

```diff
@@ -253,6 +271,159 @@ class KATANA_EXPORT RDGPartHeader {
+    // and we want to convert them to an id_dict and id_names map
+
+    // Convert EntityTypeIDs -> EntityTypeIDs map
+    tsuba::EntityTypeID cur_entity_type_id = 0;
+    tsuba::EntityTypeIDToSetOfEntityTypeIDsMap id_dict_;
+
+    for (const auto& type_set :
+         manager.GetEntityTypeIDToAtomicEntityTypeIDs()) {
+      for (tsuba::EntityTypeID i = 0; i < type_set.size(); i++) {
+        if (type_set[i] == 1) {
+          if (id_dict_.count(cur_entity_type_id)) {
+            // if we have seen this EntityTypeID already, add to its set
+            id_dict_[cur_entity_type_id].emplace_back(i);
+          } else {
+            // if we have not, create a set with the id
+            id_dict_[cur_entity_type_id] = {i};
+          }
+        }
+      }
+      cur_entity_type_id++;
+    }
+
+    // Convert EntityTypeID name map
+    tsuba::EntityTypeIDToAtomicTypeNameMap id_name_;
+    for (const auto& pair_ : manager.GetEntityTypeIDToAtomicTypeNameMap()) {
+      id_name_[pair_.first] = pair_.second;
+    }
+
+    if (!id_dict_.empty()) {
+      set_node_entity_type_id_dictionary(id_dict_);
+    }
+    if (!id_name_.empty()) {
+      set_node_entity_type_id_name(id_name_);
+    }
+  }
+
+  //TODO(emcginnis): do this generically
+  void StoreEdgeEntityTypeManager(katana::EntityTypeManager manager) {
+    // so inside we have:
+    // EntityTypeIDToAtomicTypeNameMap atomic_entity_type_id_to_type_name
+    // EntityTypeIDToSetOfEntityTypeIDsMap entity_type_id_to_atomic_entity_type_ids
+    // and we want to convert them to an id_dict and id_names map
+
+    // Convert EntityTypeIDs -> EntityTypeIDs map
+    tsuba::EntityTypeID cur_entity_type_id = 0;
+    tsuba::EntityTypeIDToSetOfEntityTypeIDsMap id_dict_;
+
+    for (const auto& type_set :
+         manager.GetEntityTypeIDToAtomicEntityTypeIDs()) {
+      for (tsuba::EntityTypeID i = 0; i < type_set.size(); i++) {
+        if (type_set[i] == 1) {
+          if (id_dict_.count(cur_entity_type_id)) {
+            // if we have seen this EntityTypeID already, add to its set
+            id_dict_[cur_entity_type_id].emplace_back(i);
+          } else {
+            // if we have not, create a set with the id
+            id_dict_[cur_entity_type_id] = {i};
+          }
+        }
+      }
+      cur_entity_type_id++;
+    }
+
+    // Convert EntityTypeID name map
+    tsuba::EntityTypeIDToAtomicTypeNameMap id_name_;
+    for (const auto& pair_ : manager.GetEntityTypeIDToAtomicTypeNameMap()) {
+      id_name_[pair_.first] = pair_.second;
+    }
+
+    if (!id_dict_.empty()) {
+      set_edge_entity_type_id_dictionary(id_dict_);
+    }
+    if (!id_name_.empty()) {
+      set_edge_entity_type_id_name(id_name_);
+    }
+  }
+
+  katana::Result<katana::EntityTypeManager> GetEntityTypeManager(
+      const tsuba::EntityTypeIDToSetOfEntityTypeIDsMap& id_dict,
+      const tsuba::EntityTypeIDToAtomicTypeNameMap& id_name) {
+    katana::EntityTypeIDToAtomicTypeNameMap manager_name_map_;
```

Comment:
trailing _ is only used for class members. Please don't use it for local variables.